packages/lit-element/src/hydrate-support.ts

```diff
@@ -30,7 +31,49 @@ interface PatchableLitElement extends HTMLElement {
 }: {
   LitElement: PatchableLitElement;
 }) => {
-  // Capture whether we need hydration or not
+  const observedAttributes = Object.getOwnPropertyDescriptor(
+    Object.getPrototypeOf(LitElement),
+    'observedAttributes'
+  )!.get!;
+
+  // Add `defer-hydration` to observedAttributes
+  Object.defineProperty(LitElement, 'observedAttributes', {
+    get() {
+      return [...observedAttributes.call(this), 'defer-hydration'];
+    },
+  });
+
+  // Enable element when 'defer-hydration' attribute is removed by calling the
+  // super.connectedCallback()
+  const attributeChangedCallback =
+    LitElement.prototype.attributeChangedCallback;
+  LitElement.prototype.attributeChangedCallback = function (
```

Comment:
I think it might be good to put as much of this logic onto ReactiveElement as possible for other subclasses to use...

Reply:
Let's do that in a later PR, since it involves adding a hydrate-support for reactive-element, which doesn't yet exist.